provider/kubernetes.go

```diff
@@ -0,0 +1,130 @@
+package provider
+
+import (
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/emilevauge/traefik/types"
+	"hash/fnv"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/util"
+	"reflect"
+)
+
+// Kubernetes holds configurations of the Kubernetes provider.
+type Kubernetes struct {
+	BaseProvider `mapstructure:",squash"`
+	Endpoint     string
+	Domain       string
+}
+
+// Provide allows the provider to provide configurations to traefik
+// using the given configuration channel.
+func (provider *Kubernetes) Provide(configurationChan chan<- types.ConfigMessage) error {
+	//var KubernetesFuncMap = template.FuncMap{}
+	var ingClient client.IngressInterface
+	config := &client.Config{
+		Host: provider.Endpoint,
+	}
```

Comment:
Maybe it's better in firstly case to try in cluster config 
`
config, err := client.InClusterConfig()
`